sklearn/utils/estimator_checks.py

```diff
@@ -3113,6 +3113,8 @@ def check_no_attributes_set_in_init(name, estimator_orig):
 
     # Test for no setting apart from parameters during init
     invalid_attr = set(vars(estimator)) - set(init_params) - set(parents_init_params)
+    # Ignore private attributes
+    invalid_attr = set([attr for attr in invalid_attr if not attr.startswith("_")])
```

Comment:
why is this needed here?

Reply:
I do not know how this got here. It was removed when I synced up this PR with main.